src/rust/engine/process_execution/src/cache.rs

```diff
@@ -206,14 +207,11 @@ impl CommandRunner {
     // (This isn't super urgent because we don't ever actually GC this store. So also...)
     // TODO: GC the local process execution cache.
 
-    let mut response_bytes = Vec::with_capacity(execute_response.encoded_len());
-    execute_response
-      .encode(&mut response_bytes)
-      .map_err(|err| format!("Error serializing execute process result to cache: {}", err))?;
+    let response_bytes = execute_response.to_bytes();
 
     let bytes_to_store = bincode::serialize(&PlatformAndResponseBytes {
       platform: result.platform,
-      response_bytes,
+      response_bytes: response_bytes.to_vec(),
```

Comment:
Maybe change the type of `response_bytes` to `Bytes` to avoid the copy here?

Reply:
I thought about that but, given this is stored in the LMDB local cache, I would then need to reason about whether there is a difference in the encoding of `PlatformAndResponseBytes` between the `Vec<u8>` and `Bytes` forms of this field. And, if there is a difference, then Pants would fail to decode users' existing cache entries which is backwards incompatible. That's too much complexity for what this PR. That conversion should be done in a separate PR.